utils.py

```diff
@@ -5,17 +5,247 @@
+        message -- explanation of the error
+    """
+
+    def __init__(self, message):
+        Exception.__init__(self)
+        self.message = message
+
+
+def io_proxy_wado(
+    api_key, api_key_prefix, project_id, study=None, series=None, instance=None
+):
+    """
+    Request wrapper for io-proxy api (https://{instance}/io-proxy/docs#/)
+
+    Args:
+        api_key (str): Full instance api-key
+        api_key_prefix (str): Type of user (e.g. 'scitran-user')
+        project_id (str): Project ID to inquire
+        study (str, optional): DICOM StudyUID. Defaults to None.
+        series (str, optional): DICOM SeriesUID. Defaults to None.
+        instance (str, optional): DICOM InstanceUID. Defaults to None.
+
+    Returns:
+        dict/list: A dictionary for dicom tags or a list of dictionaries with dicom tags
+    """
+    base_url = Path(api_key.split(":")[0])
+    base_url /= "io-proxy/wado"
+    base_url /= "projects/" + project_id
+    if study:
+        base_url /= "studies/" + study
+    if series:
+        base_url /= "series/" + series
+        base_url /= "instances"
+    if instance:
+        base_url /= instance
+        base_url /= "tags"
+    base_url = "https://" + str(base_url)
+
+    headers = {
+        "Authorization": api_key_prefix + " " + api_key,
+        "accept": "application/json",
+    }
+
+    req = requests.get(base_url, headers=headers)
+
+    return leval(req.text)
+
+
+def convert_ROI_to_nifti_form(
+    fw_client, project_id, file_obj, imagePath, ohifViewer_info
+):
+    """
+    Converts ohifViewer ROI from a dicom to that of an individual nifti format.
+
+    Args:
+        fw_client (flywheel.Client): The active flywheel client
+        project_id (str): The id of the project to search for io-proxy elements
+        file_obj (flywheel.File): File object data
+        imagePath (str): The DICOM representation of the ohifViewer ROI imagePath
+        ohifViewer_info (dict): The full session-level ohifViewer ROI dictionary.
+
+    Returns:
+        dict: File object (file_obj) dictionary
+    """
+    host = fw_client._fw.api_client.configuration.host[:-8]
+    api_key_prefix = fw_client._fw.api_client.configuration.api_key_prefix[
+        "Authorization"
+    ]
+    api_key_hash = fw_client._fw.api_client.configuration.api_key["Authorization"]
+    api_key = ":".join([host.split("//")[1], api_key_hash])
+
+    out_ohifViewer_info = {"measurements": {}}
+
+    study_id, series_id = imagePath.split("$$$")[:2]
+
+    instances = io_proxy_wado(api_key, api_key_prefix, project_id, study_id, series_id)
+
+    for roi_type in ["FreehandRoi", "RectangleRoi", "EllipticalRoi"]:
+        if ohifViewer_info["measurements"].get(roi_type):
+            for roi in ohifViewer_info["measurements"].get(roi_type):
+                if imagePath in roi["imagePath"]:
+                    instance_id = roi["imagePath"].split("$$$")[2]
+                    slice_instance = [
```

Comment:
So this gets the good stuff?  Maybe include a description of what is being gotten?

Reply:
Good idea!!